lblod/ember-rdfa-editor
# Update to latest ember-appuniversum v3.4.0

### Overview
Passing icons as components instead of strings, as this uses inline SVGs rather than making an extra request for an svgiconset which fails on embeddable.

This allows and requires the latest ember-appuniversum, 3.4.0, as without that, the passing of components to `@icon` arguments will fail. Any usage of icons using a `string` argument will result in the svgiconset version being used, which will fail on embeddable.

In order to support importing icons in template only components, added support for `.gts` single file templates and updated the tooling to add better type checking supported by these.

##### connected issues and PRs:
Part of https://github.com/lblod/frontend-embeddable-notule-editor/issues/243
Uses the work on https://github.com/appuniversum/ember-appuniversum/issues/482

### Setup
You might need to do some set-up in your editor if you haven't set it up to support the new template syntax, details [in the ember docs](https://guides.emberjs.com/release/components/template-tag-format/#toc_editor-integrations) for highlighting and [the glint docs](https://typed-ember.gitbook.io/glint/getting-started#glint-editor-extensions) for type checking.

### How to test/reproduce
All the icons should work. If you inspect an icon, you should no longer see a `<use>` element inside the `<svg>`, you should just see the svg inline.

### Challenges/uncertainties
While refactoring it was easier to write macros that relied on `'` quotes being used instead of optionally supporting `"`, so I converted them. This results in more consistency between files anyway.

There's a bug with some of the new icon components, noted [here](https://github.com/appuniversum/ember-appuniversum/pull/488) which I have worked around rather than waiting for a new AU release.

### Checks PR readiness
